test/test_permissions.py

```diff
@@ -0,0 +1,12 @@
+"""Tests for permissions infrastructure."""
+
+import conftest
+
+
+def test_require_reader(server, managed_workspace, managed_user):
+    """Test the `require_reader` decorator."""
+    with conftest.login(managed_user, server):
+        resp = server.get(f"/api/workspaces/{managed_workspace}")
+
+    assert resp.status_code == 200
+    assert resp.json["permissions"]["owner"] == managed_user.sub
```

Comment:
I think we should be testing all of the `require_*` decorators, but I'm okay with pushing that off to a future PR.

Reply:
Definitely agree. I opened #428 to address that, but still want to get this out the door asap, since it's holding up a slew of other work atm.